src/main/java/org/prebid/server/analytics/AnalyticsReporterDelegator.java

```diff
@@ -168,4 +181,43 @@ private static ObjectNode prepareAnalytics(ObjectNode analytics, String adapterN
 
         return !analyticsNodeCopy.isEmpty() ? analyticsNodeCopy : null;
     }
+
+    private <T> void processEventWithResult(T event, AnalyticsReporter analyticsReporter) {
+        final String analyticsCode = analyticsReporter.name();
+        analyticsReporter.processEvent(event)
+                .compose(s -> processSuccess(s, analyticsCode))
```

Comment:
You may use `map` instead of `compose` since you don't have to return `Future`.
Also, we can original event: `.map(ignored -> processSuccess(event, reporterName))`.